### PR TITLE
Add libffi 3.3-rc0

### PR DESCRIPTION
**Pull request by EmilioPeJu**
The libffi library provides a portable, high level programming interface to various calling conventions

### CODE REVIEW

File: packages/libffi/CONFIG [hunk 1]
```diff
@@ -0,0 +1,25 @@
+libffi_VERSION ?= 3.3-rc0
```
Comment (Araneidae) [inline packages/libffi/CONFIG]: Why version 3.3, particularly as we only have an rc0 candidate?  I'm guessing there's a reason 3.2.1 isn't suitable.

File: packages/libffi/CONFIG [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+TAR_EXTENSION = .tar.gz
+
+PATCHES_3.3-rc0 = \
+        0001-Fix-installation-location-of-libffi.patch \
+        0002-Fix-use-of-compact-eh-frames-on-MIPS.patch \
+        0003-libffi-enable-hardfloat-in-the-MIPS-assembly-code.patch
```
Comment (Araneidae) [inline packages/libffi/CONFIG]: (Utterly trivial, but four space indent is more consistent with usage elsewhere.)

File: packages/libffi/patches/0001-Fix-installation-location-of-libffi.patch [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ MAKEOVERRIDES=
+ 
+-toolexeclib_LTLIBRARIES = libffi.la
++lib_LTLIBRARIES = libffi.la
+ noinst_LTLIBRARIES = libffi_convenience.la
+ 
+ libffi_la_SOURCES = src/prep_cif.c src/types.c \
+-- 
+2.5.3
+
```
Comment (Araneidae) [inline packages/libffi/patches/0001-Fix-installation-location-of-libffi.patch]: I wonder if this patch in particular could be pushed upstream?

**Review by Araneidae — COMMENTED** [review]
Three general comments.
- I've realised that we've not been recording the origin of our sources consistently.  It would be good to point to https://sourceware.org/libffi/ for this one
- I'm in two minds about the MIPS patches.  On the one hand they are clearly part of the buildroot patch set, on the other hand we don't need them, and in particular patch #3 bothers me with its comment "... temporary fix ... needs to be fixed"!
- We definitely need to credit the source of the patches.

File: packages/libffi/CONFIG (hunk 1 above)
Comment (EmilioPeJu) [reply packages/libffi/CONFIG]: I assumed the patches were made for that version, but if you want, I could try  rc2.
Regarding version 3.2.1, it's more than 5 years old, so I guess there are major changes that are worth getting.

File: packages/libffi/CONFIG (hunk 2 above)
Comment (EmilioPeJu) [reply packages/libffi/CONFIG]: Do you want me to drop the last 2 patches?

File: packages/libffi/patches/0001-Fix-installation-location-of-libffi.patch (hunk 3 above)
Comment (EmilioPeJu) [reply packages/libffi/patches/0001-Fix-installation-location-of-libffi.patch]: We are using packageconfig so I suspect this patch doesn't affect us

**Comment by EmilioPeJu** [comment]
I've simplified a bit

**Comment by Araneidae** [comment]
No patches!  Think I approve of that.

**Comment by Araneidae** [comment]
I will merge this, but I think we should step to the final 3.3 release as soon as we can.

**Pull request merged**